pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1443,10 +1443,10 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                         .exceptionally((ex) -> {
                                             log.warn("Replication or dedup check failed."
                                                     + " Removing topic from topics list {}, {}", topic, ex);
-                                            persistentTopic.stopReplProducers().whenComplete((v, exception) -> {
```

Comment:
Does the line 1435 need to do the same thing? `persistentTopic.stopReplProducers().whenComplete((v, exception)`

Reply:
We don't need change here, because it already completed with exception. the write lock is released.